controller/token.go

```diff
@@ -342,41 +343,104 @@ func (c *TokenController) Exchange(ctx *app.ExchangeTokenContext) error {
 	if payload == nil {
 		return jsonapi.JSONErrorResponse(ctx, errors.NewBadParameterError("payload", "nil").Expected("not empty payload"))
 	}
-	if payload.ClientID == nil {
-		return jsonapi.JSONErrorResponse(ctx, errors.NewBadParameterError("client_id", "nil").Expected("Service Account ID"))
-	}
-	if payload.ClientSecret == nil {
-		return jsonapi.JSONErrorResponse(ctx, errors.NewBadParameterError("client_secret", "nil").Expected("Service Account secret"))
-	}
+	if payload.GrantType == "client_credentials" {
```

Comment:
add this string as a constant at the top of this file?

Reply:
https://github.com/fabric8-services/fabric8-auth/pull/210/commits/391604b2affa2b852131a55a5f94a7d1b120c10c